js/adapt-contrib-glossary.js

```diff
@@ -53,7 +52,9 @@ define(function(require) {
         Adapt.drawer.addItem(drawerObject, 'glossary:showGlossary');
 
         setupGlossary(courseGlossary, courseGlossary._glossaryItems);
+    }
 
-    });
+    Adapt.once('app:dataReady', initGlossary);
+    Adapt.on('languagePicker:languageChange', initGlossary);
```

Comment:
If you do this, I think you'll run into the same problem we had with Resources in that it can get added to the Drawer twice under certain conditions - you might want to change this to something more like [this](https://github.com/adaptlearning/adapt-contrib-resources/pull/43/commits/7a09f528b2107e723e79db33904d94be2554f0eb)